MetaMask/metamask-extension
# feat: UX: Multichain: Switch Networks on Popup Open

## **Description**
This PR aims to switch the networks upon the MetaMask popup opening in different tabs.  When the network is switched, provide a toast to the user saying what network was changed to.  Closes other MetaMask instances upon change.

*Note:* Closing other popup windows will be done in another PR.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1779

## Caveats

1.  The "Network Info" popup may display over the home screen, which is annoying.  We're going to make a decision about that before release, likely removing it.
2. There may be a flicker on the switched-network Toast. We'll be improving that separately.

## **Manual testing steps**

### Manual test 1
1. Go to https://app.uniswap.org, connect your wallet and choose Polygon
3. In another tab, open https://pancakeswap.finance/ and switch to BNB Chain
4. Go back to the Uniswap tab
5. Open MetaMask, see Polygon as selected network
6. Go back to the PancakeSwap tab
7. 4. Open MetaMask, see BNB Chain as selected network

### Manual test 2 (builds off 1)
1.  Go to https://metamask.github.io/test-dapp/ - should switch you to Polygon
2. Click the "Send" button (this creates a confirmation), wait for confirmation to open
3. Ignore confirmation, switch back to Pancake Swap tab
4. Click the MetaMask extension icon
5. See in the top right of the confirmation that the Network still shows Polygon (you haven't switched to BNB chain yet)
6. Cancel/reject the Send
8. See the network auto-change to BNB chain

## **Screenshots/Recordings**

### **Before**

N/A

### **After**


https://github.com/MetaMask/metamask-extension/assets/46655/f7968d87-5dbe-4205-b6cb-2771e4169770




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
